src/promptflow/promptflow/core/_serving/app.py

```diff
@@ -0,0 +1,253 @@
+# ---------------------------------------------------------
```

Comment:
A small question, may not related to this pr. 

![image](https://github.com/microsoft/promptflow/assets/2572521/d1d6264d-1ffe-495f-8e61-ceb1aa152fb8)
![image](https://github.com/microsoft/promptflow/assets/2572521/d1e81ea8-6608-457e-8502-b5537e25bdc9)

I don't have preference using `Core` as suffix or prefix, but we'd better have the same naming style.

cc @brynn-code  @crazygao

Reply:
This class is not exported, I'll adapt to _CoreXXX next time.

Reply:
For connection, the imported class can't be named as 'ConnectionCore'. because there is some calculation based on public class, aka class name with no '_' prefix. I'll vote for _CoreXXX or _XXXCore